### PR TITLE
[MERGE] web,*: allow searchpanel in all multi-record views

### PR DESCRIPTION
**Pull request by aab-odoo**
*base,hr,lunch

From now on, search panels can be activated on any multi-record views, and are no longer
restricted to kanban views.

The arch of the search panel has to be specified in the search view.

HR and lunch usecases have been adapted accordingly.

Task 1985921

### CODE REVIEW

File: addons/web/static/src/js/views/abstract_view.js [hunk 1]
```diff
@@ -150,6 +155,7 @@ var AbstractView = Factory.extend({
             modelName: params.modelName,
             res_id: currentId,
             res_ids: controllerState.resIds || params.ids || (currentId ? [currentId] : undefined),
+            viewType: this.viewType,
```
Comment (kebeclibre) [inline addons/web/static/src/js/views/abstract_view.js]: seems deprec

**Comment by aab-odoo** [comment]
robodoo rebase-ff

**Comment by aab-odoo** [comment]
robodoo rebase-merge

File: addons/web/static/src/scss/search_panel.scss [hunk 2]
```diff
@@ -0,0 +1,94 @@
+// ------- View with SearchPanel -------
+$o-searchpanel-p: $o-horizontal-padding;
+$o-searchpanel-p-small: $o-horizontal-padding*0.5;
+$o-searchpanel-p-tiny: $o-searchpanel-p-small*0.5;
+
+$o-searchpanel-category-default-color: $o-brand-primary;
+$o-searchpanel-filter-default-color: #D59244;
```
Comment (VincentSchippefilt) [inline addons/web/static/src/scss/search_panel.scss]: duplicate with o-dms-color-tag

File: doc/reference/views.rst [hunk 3]
```diff
@@ -1972,6 +1920,64 @@ assuming ``foo`` is a field and ``bar`` is a filter an action context of:
 will automatically enable the ``bar`` filter and search the ``foo`` field for
 *acro*.
 
+``searchpanel``
```
Comment (aab-odoo) [inline doc/reference/views.rst]: move above

**Comment by VincentSchippefilt** [comment]
robodoo r+

**Comment by Bonainifo** [comment]
SearchPannet can not work in lunch & hr or other modules. Add searchpannel section into views, always get a error “got an unexpected keyword argument 'env'

**Comment by aab-odoo** [comment]
Hello @Bonainifo 

Could you give more details? Thanks.

**Comment by Guobower** [comment]
Sorry, I creat a new database and install lunch, now it work properly. The error message appear after I edit another list view. I will try it again

**Pull request closed without merging**